lib/saml_idp/xml_security.rb

```diff
@@ -55,9 +55,17 @@ def validate(idp_cert_fingerprint, soft = true)
         # check cert matches registered idp cert
         fingerprint = fingerprint_cert(cert)
         sha1_fingerprint = fingerprint_cert_sha1(cert)
-        plain_idp_cert_fingerprint = idp_cert_fingerprint.gsub(/[^a-zA-Z0-9]/,"").downcase
 
-        if fingerprint != plain_idp_cert_fingerprint && sha1_fingerprint != plain_idp_cert_fingerprint
+        fingerprint_validation = false
+
+        [idp_cert_fingerprint].flatten.each do |idp_cert_fingerprint_single|
+          plain_idp_cert_fingerprint = idp_cert_fingerprint_single.gsub(/[^a-zA-Z0-9]/,"").downcase
+          if fingerprint == plain_idp_cert_fingerprint || sha1_fingerprint != plain_idp_cert_fingerprint
```

Comment:
This looks like it was an `&&` before, now it's a `||`, is that right? The `|| sha1_fingerprint != plain_idp_cert_fingerprint` looks like it will be true a lot of the time.

Reply:
On second look, I think the second `!=` should be `==` and the rest is correct. What do you think?